### DISCUSSION TITLE
Hello, Can You Help me - Material UI - Select - Not have OnChange

### DISCUSSION THREAD

**Discussion by albertolotz** [react-hook-form/react-hook-form#2373] [Q&A] [answered]
I have the code, as follow ... I need where change a funcion will be dipared, but the event OnChange doen't work, if I use OnClik it work as expected . Tks
<FormControl
                className={classes.formControl}
                margin="normal"
                variant="outlined"
                defaultValue=""
              >
                <InputLabel id="label-uf">UF</InputLabel>
                <Controller
                  label="UF"
                  as={
                    <Select
                      onChange={mudar}
                      onWheel={handleChange}
                      labelId="label-uf"
                      required
                      value={uf || ''}
                    >
                      <MenuItem value="">
                        <em>Vazio</em>
                      </MenuItem>
                      {uf.map(item => (
                        <MenuItem key={item.value} value={item.value}>
                          {item.label}
                        </MenuItem>
                      ))}
                    </Select>
                  }
                  control={control}
                  name="uf"
                />
              </FormControl>

**Comment by bluebill1049** [comment]
there is an example on the doc: https://react-hook-form.com/api#Controller

https://codesandbox.io/s/react-hook-form-v6-controller-qsd8r

**Reply by albertolotz** [reply]
Tks for help.
I tryed, but using render the Select control have a problem in Label and return value undefined .

**Comment by neighborhood999** [comment]
> I have the code, as follow ... I need where change a funcion will be dipared, but the event OnChange doen't work, if I use OnClik it work as expected . Tks
> 
> UF
> <Controller
> label="UF"
> as={
> <Select
> onChange={mudar}
> onWheel={handleChange}
> labelId="label-uf"
> required
> value={uf || ''}
> >
> 
> _Vazio_
> 
> {uf.map(item => (
> 
> {item.label}
> 
> ))}
> 
> }
> control={control}
> name="uf"
> />

I think you can indent your code, it's hard to read...

**Comment by albertolotz** [comment, marked as answer]
![Captura de tela de 2020-07-26 22-58-53](https://user-images.githubusercontent.com/55667164/88496579-ff625d00-cf93-11ea-9be0-6bc5ce628aaf.png)
![Captura de tela de 2020-07-26 22-59-28](https://user-images.githubusercontent.com/55667164/88496590-0a1cf200-cf94-11ea-9789-b678f044eb0c.png)

<FormControl
                className={classes.formControl}
                margin="normal"
                variant="outlined"
                defaultValue=""
              >
                <InputLabel id="label-uf">UF</InputLabel>
                <Controller
                  label="UF"
                  control={control}
                  name="uf"
                  render={props => (
                    <Select
                      onChange={e => handleChange(e)}
                      labelId="label-uf"
                      required
                      value={ufSelect || ''}
                    >
                      <MenuItem value="">
                        <em>Vazio</em>
                      </MenuItem>
                      {uf.map(item => (
                        <MenuItem key={item.value} value={item.value}>
                          {item.label}
                        </MenuItem>
                      ))}
                    </Select>
                  )}
                />
              </FormControl>

**Reply by albertolotz** [reply]
look a label um UF control, it is over the line

**Reply by bluebill1049** [reply]
onChange={e => handleChange(e)}

console.log(e) see what's in there and try to indent code, it's bit hard to read.

**Reply by albertolotz** [reply]
Sorry for the code, not indent. I copy e past but when submit it self align in left. it's my first help request on this channel sorry, sorry! 
I solved the problem . the last, is about de **label in UF,** as you can see on the picture de line pass over de text, I tried many ways and doesn't solve. 

Last, thanks for you time! this Lib is amazing congratulations to you and the team!

![Captura de tela de 2020-07-26 23-28-01](https://user-images.githubusercontent.com/55667164/88497996-4e11f600-cf98-11ea-88de-a74adee686dc.png)

**Reply by bluebill1049** [reply]
nice glad you work it out 👍 no worries we all have our first time. tips for you: indent is easy

you need the ``` in from your code.


and ``` to close your code.

free indentation.
